web-access-control/protected-operation/read-access-public.feature

```diff
@@ -0,0 +1,115 @@
+Feature: Public agents can read (and only that) a resource when granted read access
+  Background: Create test resources with correct access modes
+    * def authHeaders = (method, url, public) => !public ? clients.bob.getAuthHeaders(method, url) : {}
+    * def createResources =
+    """
+      function (modes) {
+        const testContainer = rootTestContainer.reserveContainer()
+        const plainResource = testContainer.createResource('.txt', 'Hello', 'text/plain')
+        plainResource.accessDataset = plainResource.accessDatasetBuilder.setPublicAccess(plainResource.url, modes).build()
+        const rdfResource = testContainer.createResource('.ttl', karate.readAsString('../fixtures/example.ttl'), 'text/turtle')
+        rdfResource.accessDataset = rdfResource.accessDatasetBuilder.setPublicAccess(rdfResource.url, modes).build()
+        const container = testContainer.createContainer()
+        container.accessDataset = container.accessDatasetBuilder.setPublicAccess(container.url, modes).build()
+        return { plain: plainResource, rdf: rdfResource, container: container }
```

Comment:
I realize this has something to do with the lacking UMA support, but I just can't see how the setup here differs from the `read-access-agent.feature` and so I don't understand how this will produce a different outcome.

Reply:
The bob vs agent vs public setups are based on granting access to bob, or any authenticated agent, or any public agent (which would need UMA support in an ACP context). The 3 feature file setups therefore only differ on the access calls:
* setAgentAccess
* setAuthenticatedAccess
* setPublicAccess
It is a very small difference so easy to miss. I wondered about combining tests and using a flag to determine which mode to use but it got very complex. Since the access tables in each feature file differ we need the 3 files per set. So could I extract this function to a shared feature and call it as an external function? Sadly not, the setup steps are then hidden in the report and it is really important that we can see the ACL documents.

Reply:
Ah, right! Yeah, I missed that. 

Do you think it is appropriate (and possible) to have a comment in the code to point that out?

Reply:
That would certainly help